src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -322,7 +321,11 @@ public static KNNQueryBuilder fromXContent(XContentParser parser) throws IOExcep
             }
         }
 
-        validateSingleQueryType(k, maxDistance, minScore);
+        VectorQueryType vectorQueryType = validateSingleQueryType(k, maxDistance, minScore);
```

Comment:
Probably we can rename the method and move increment operation local to that function validated of any of the parameters are not null and set the enum as well the increment.

Reply:
It helps to localize the logic.

Reply:
@Vikasht34 Thanks for the comment, I had talked with @navneet1v on this part on original PR, if we separate out to validateSingleQuery and getQueryType, there will have duplicate code between the two methods. So I keep it as single method here for now.